examples/hosts/app-integration/external-data/src/view/debugView.tsx

```diff
@@ -158,14 +156,151 @@ function debugResetExternalData(): void {
 // themselves (as if they were editing it outside of Fluid).
 // TODO: Consider how we might simulate errors/failures here to play with retry and recovery.
 const ControlsView: React.FC<IControlsViewProps> = (props: IControlsViewProps) => {
-    console.log(props.model);
     return (
         <div>
             <h3>Debug controls</h3>
             <div style={{ margin: "10px 0" }}>
                 <button onClick={ debugResetExternalData }>Reset external data</button><br />
-                <button onClick={ props.model.debugSendCustomSignal }>Trigger external data change signal</button><br />
             </div>
         </div>
     );
 };
+
+
+interface ITaskRowProps {
+    task: ExternalDataTask
+}
+
+/**
+ * The view for a single task in the TaskListView, as a table row.
+ */
+const TaskRow: React.FC<ITaskRowProps> = (props: ITaskRowProps) => {
+    const { task } = props;
+
+    const IdChangeHandler = (e: React.SyntheticEvent<HTMLInputElement>): void => {
+        task.id = e.currentTarget.value;
+    };
+    const NameChangeHandler = (e: React.SyntheticEvent<HTMLInputElement>): void => {
+        task.name = e.currentTarget.value;
+    };
+    const PriorityChangeHandler = (e: React.SyntheticEvent<HTMLInputElement>): void => {
+        task.priority =  Number.parseInt(e.currentTarget.value, 10);
+    };
+
+    return (
+        <tr>
+            <td><input defaultValue={ task.id }  style={{ width: "30px" }} onChange={ IdChangeHandler }></input></td>
+            <td><input defaultValue={ task.name } style={{ width: "200px" }} onChange={ NameChangeHandler }></input></td>
+            <td><input defaultValue={ task.priority } type="number" style={{ width: "50px" }} onChange={ PriorityChangeHandler }></input></td>
+        </tr>
+    );
+};
+
+class ExternalDataTask {
```

Comment:
Nit: Does this need to be a class? Or can it just be an interface?

Reply:
I turned it into a class so that I could pass it into the TaskRow component and manipulate the values from the TaskRow component. Can I do that if it were only an interface?

Reply:
Not blocking for the PR, but I think yes? And instead of constructing an `ExternalDataTask` on line 234 you could just return an anonymous object as long as it has the three properties.

Reply:
All righty! Got it! Changed it! It works!